File: scenario/spot.py
import numpy as np

class Spot:
    """Class for standing spots."""
    def __init__(self, x, y):
        self.x = x
        self.y = y

    def utility(self, x):
        """Calculates the utility of the spot"""
        distance = abs(self.x - x)
        return np.e ** (-0.5 * (distance / 2) ** 2)

File: cpp/scenario/spot.cpp
#include "spot.h"

#include <cmath>


Spot::Spot(double spot_x, double spot_y) {
    x = spot_x;
    y = spot_y;
}

double Spot::utility(double px) {
    double distance = std::abs(x - px);
    return exp(-0.5*pow(distance/2, 2));
}
